contracts/BaseLaunchpeg.sol

```diff
@@ -360,6 +431,76 @@ abstract contract BaseLaunchpeg is
         emit DevMint(msg.sender, _quantity);
     }
 
+    /// @dev Should only be called in the pre-mint phase
+    /// @param _quantity Quantity of NFTS to mint
+    function _preMint(uint256 _quantity) internal {
+        if (_quantity == 0) {
+            revert Launchpeg__InvalidQuantity();
+        }
+        if (_quantity > allowlist[msg.sender]) {
+            revert Launchpeg__NotEligibleForAllowlistMint();
+        }
+        if (
+            totalSupply() + _quantity > collectionSize ||
+            amountMintedDuringAllowlist + _quantity > amountForAllowlist
+        ) {
+            revert Launchpeg__MaxSupplyReached();
+        }
+        allowlist[msg.sender] -= _quantity;
+        uint256 price = _preMintPrice();
+        uint256 totalCost = price * _quantity;
+        numberPreMinted[msg.sender] += _quantity;
+        amountMintedDuringPreMint += _quantity;
+        amountMintedDuringAllowlist += _quantity;
```

Comment:
this seems weird to update `amountMintedDuringAllowlist` during the pre-mint phase no?

can we change `amountMintedDuringAllowlist + _quantity > amountForAllowlist` just above into `amountMintedDuringPreMint + _quantity > amountForAllowlist`? 

This way, we'll still have `amountMintedDuringAllowlist` to know how many NFTs were minted during the actual AL phase and `amountMintedDuringPreMint` for the pre-mint phase

It would also mean we have to verify for `amountMintedDuringAllowlist + amountMintedDuringPreMint + _quantity > amountForAllowlist` in the method `allowlistMint` in both flat and auction contracts

What do you think?

Reply:
updated. i realised while updating this that i didn't correctly account for the pre-minted amount in the checks for some of the mint methods to avoid going over the collection size. i added 2 internal methods `_totalSupplyWithPreMint()` and `_numberMintedWithPreMint()` to handle this. let me know if you have any thoughts about this

Reply:
yeah I see good catch, looks good to me